ecs-init/docker/docker.go

```diff
@@ -440,6 +463,33 @@ func (c *client) getHostConfig(envVarsFromFiles map[string]string) *godocker.Hos
 	return createHostConfig(binds)
 }
 
+// getCredentialsFetcherSocketBind returns the bind for CredentialsFetcher socket and also whether the bind exists.
+// Value for the bind is as follow:
+// CREDENTIALS_FETCHER_HOST (as in os.Getenv) set: source CREDENTIALS_FETCHER_HOST (as in os.Getenv, trim unix:// prefix),
+// dest CREDENTIALS_FETCHER_HOST (as in /etc/ecs/ecs.config, trim unix:// prefix)
+func getCredentialsFetcherSocketBind(envVarsFromFiles map[string]string) (string, bool) {
+	CredentialsFetcherEndpointAgent := defaultCredentialsFetcherSocketPath
+	CredentialsFetcherUnixSocketSourcePath, fromEnv := config.CredentialsFetcherUnixSocket()
+	if fromEnv {
+		credentialsFetcherEndpointConfig, ok := envVarsFromFiles[config.CredentialsFetcherHostEnvVar]
+		if ok && strings.HasPrefix(credentialsFetcherEndpointConfig, config.UnixSocketPrefix) {
+			CredentialsFetcherEndpointAgent = strings.TrimPrefix(credentialsFetcherEndpointConfig, config.UnixSocketPrefix)
+		} else {
+			CredentialsFetcherEndpointAgent = defaultCredentialsFetcherSocketPath
+		}
+	}
+
+	// check whether the path to the mount exists
+	_, err := os.Stat(CredentialsFetcherUnixSocketSourcePath)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
Why is this being checked in multiple places?

Reply:
one check on the init and other on the agent as well, to check the socket is properly bind mount